src/Task/Transfer/ScpTask.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace TYPO3\Surf\Task\Transfer;
+
+/*
+ * This file is part of TYPO3 Surf.
+ *
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+use TYPO3\Flow\Utility\Files;
+use TYPO3\Surf\Domain\Model\Application;
+use TYPO3\Surf\Domain\Model\Deployment;
+use TYPO3\Surf\Domain\Model\Node;
+use TYPO3\Surf\Domain\Model\Task;
+use TYPO3\Surf\Domain\Service\ShellCommandServiceAwareInterface;
+use TYPO3\Surf\Domain\Service\ShellCommandServiceAwareTrait;
+use TYPO3\Surf\Exception\InvalidConfigurationException;
+use TYPO3\Surf\Exception\TaskExecutionException;
+
+/**
+ * A scp transfer task
+ *
+ * Copies the application assets from the application workspace to the node using scp.
+ */
+final class ScpTask extends Task implements ShellCommandServiceAwareInterface
+{
+    use ShellCommandServiceAwareTrait;
+
+    /**
+     * @param Node $node
+     * @param Application $application
+     * @param Deployment $deployment
+     * @param array $options
+     *
+     * @throws InvalidConfigurationException
+     */
+    public function execute(Node $node, Application $application, Deployment $deployment, array $options = [])
+    {
+        $fileName = sprintf('%s.tar.gz', $deployment->getReleaseIdentifier());
+
+        $localPackagePath = $deployment->getWorkspacePath($application);
+        $releasePath = Files::concatenatePaths([$application->getReleasesPath(), $deployment->getReleaseIdentifier()]);
+
+        // Create remote transfer path if not exist
+        $remoteTransferPath = Files::concatenatePaths([$application->getDeploymentPath(), 'cache', 'transfer']);
+        $this->shell->executeOrSimulate(sprintf('mkdir -p %s', $remoteTransferPath), $node, $deployment);
+
+        // Create the scp destination command
+        $destinationArgument = $node->isLocalhost() ? $remoteTransferPath : sprintf('%s@%s:%s', $node->getUsername(), $node->getHostname(), $remoteTransferPath);
+
+        // escape whitespaces
+        $localPackagePath = preg_replace('/\s+/', '\ ', $localPackagePath);
+        $destinationArgument = preg_replace('/\s+/', '\ ', $destinationArgument);
+
+        // Create a localhost node and create tarball on it
+        $localhost = new Node('localhost');
+        $localhost->onLocalhost();
+        // To prevent issues remove all tar.gz in it before
+        $this->shell->executeOrSimulate(sprintf('rm -rf %s/*.tar.gz', $localPackagePath), $localhost, $deployment);
+        // Create tarball locally in the workspace directory
+        $this->shell->executeOrSimulate(sprintf('cd %1$s/; tar %3$s -zcf %2$s -C %1$s .', $localPackagePath, $fileName, $this->getExcludes($options)), $localhost, $deployment, false, false);
```

Comment:
I get an error here on my Ubuntu 18 machine (surprisingly not always but in 99%):
```
(localhost): "cd /home/xxx/.surf/workspace/live/xxx/ && tar --exclude=".git" -zcf 20200202191056.tar.gz -C /home/xxx/.surf/workspace/live/xxx ."

Got exception "Command returned non-zero return code: 1" but rollback disabled. Stopping.
```

Reply:
When I extracted the tar.gz I noticed that the file itself is also included.